src/main/java/com/example/DataService/CurrencyService.java

```diff
@@ -0,0 +1,36 @@
+package com.example.DataService;
+
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.example.repository.CurrencyRepository;
+
+import model.Asset;
+import model.Currency;
+
+@Service
+public class CurrencyService {
+	
+	private CurrencyRepository cr;
+	
+	public CurrencyService(CurrencyRepository cr) {
+		this.cr = cr;
+	}
+	
+	public void putCurrencies(){
+
+		final String uri = "https://api.coincap.io/v2/assets/";
+	    RestTemplate restTemplate = new RestTemplate();
+	    Asset result = restTemplate.getForObject(uri, Asset.class);
```

Comment:
Have you considered error states? What if this url returns http statuses like 4xx or 5xx?